hathor/consensus/poa/poa_block_producer.py

```diff
@@ -164,12 +164,7 @@ def _produce_block(self, previous_block: PoaBlock) -> None:
     def _expected_block_timestamp(self, previous_block: Block) -> int:
         """Calculate the expected timestamp for a new block."""
         height = previous_block.get_height() + 1
-        is_in_turn = poa.is_in_turn(settings=self._poa_settings, height=height, signer_index=self._signer_index)
-        timestamp = previous_block.timestamp + self._settings.AVG_TIME_BETWEEN_BLOCKS
-        if is_in_turn:
-            return timestamp
-
-        signer_count = len(self._poa_settings.signers)
-        assert signer_count >= 1
-        random_offset = self.manager.rng.choice(range(signer_count * _RANDOM_DELAY_MULTIPLIER)) + 1
-        return timestamp + random_offset
+        expected_index = poa.in_turn_signer_index(settings=self._poa_settings, height=height)
+        index_distance = self._signer_index - expected_index
```

Comment:
I guess you should do the `% len(...)` here (instead of the in the next line), and also add an assert that `0 <= index_distance < len(...)`.

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/1074/commits/4ce9bf815771d084eebb67121cb19037b312b025